great_expectations/rule_based_profiler/data_assistant_result/data_assistant_result.py

```diff
@@ -1554,373 +1594,351 @@ def _get_interactive_expect_column_values_to_be_between_chart(
-                & (alt.datum.max_value <= alt.datum[metric_plot_component.name])
-            )
-        elif strict_max:
-            predicate = (
-                (alt.datum.min_value >= alt.datum[metric_plot_component.name])
-                & (alt.datum.max_value > alt.datum[metric_plot_component.name])
-            ) | (
-                (alt.datum.min_value <= alt.datum[metric_plot_component.name])
-                & (alt.datum.max_value < alt.datum[metric_plot_component.name])
-            )
-        else:
-            predicate = (
-                (alt.datum.min_value >= alt.datum[metric_plot_component.name])
-                & (alt.datum.max_value >= alt.datum[metric_plot_component.name])
-            ) | (
-                (alt.datum.min_value <= alt.datum[metric_plot_component.name])
-                & (alt.datum.max_value <= alt.datum[metric_plot_component.name])
-            )
-
-        if sequential:
-            return DataAssistantResult._get_interactive_detail_expect_column_values_to_be_between_line_chart(
-                expectation_type=expectation_type,
-                df=df,
-                metric_plot_component=metric_plot_component,
-                batch_plot_component=batch_plot_component,
-                domain_plot_component=domain_plot_component,
-                min_value_plot_component=min_value_plot_component,
-                max_value_plot_component=max_value_plot_component,
-                strict_min_plot_component=strict_min_plot_component,
-                strict_max_plot_component=strict_max_plot_component,
-                predicate=predicate,
-            )
-        else:
-            return DataAssistantResult._get_interactive_expect_column_values_to_be_between_bar_chart(
-                expectation_type=expectation_type,
-                df=df,
-                metric_plot_component=metric_plot_component,
-                batch_plot_component=batch_plot_component,
-                domain_plot_component=domain_plot_component,
-                min_value_plot_component=min_value_plot_component,
-                max_value_plot_component=max_value_plot_component,
-                strict_min_plot_component=strict_min_plot_component,
-                strict_max_plot_component=strict_max_plot_component,
-                predicate=predicate,
-            )
+        return alt.layer(*lines_and_points_list)
 
     @staticmethod
-    def _get_line_chart(
+    def _get_bar_chart(
         df: pd.DataFrame,
-        metric_plot_component: MetricPlotComponent,
+        metric_plot_components: List[MetricPlotComponent],
         batch_plot_component: BatchPlotComponent,
         domain_plot_component: DomainPlotComponent,
-    ) -> alt.Chart:
+    ) -> alt.LayerChart:
         title: alt.TitleParams = determine_plot_title(
-            metric_plot_component=metric_plot_component,
+            metric_plot_components=metric_plot_components,
```

Comment:
I could envision a helper method called `determine_plot_title_and_generate_tooltip`, but all it would do is call these two other helper methods we have defined already (`determine_plot_title` and `generate_tooltip`). I'm not crazy about doing that just for the sake of saving a few lines and upon closer reflection little to no lines would. be saved since the `PlotComponent`s for each helper would need to be explicitly passed depending on the chart type.

Reply:
👍